pkg/model/application.proto

```diff
@@ -35,6 +35,7 @@ message Application {
     // What kind of this application.
     ApplicationKind kind = 6 [(validate.rules).enum.defined_only = true];
     // The path to the git directory of this application.
+    // NOTE: This field is immutable.
```

Comment:
Does that mean this field `should be unchanged` or we just `unable to make changes on this field` due to some guarantee mechanism? 👀

Reply:
It just means we must not change this field. But it should be kept immutable by using some kinds of guarantee mechanism.

Reply:
ah, I see 👍 Then instead of `NOTE`, make it `TODO:` may be better suited to this issue, we should look up at this later 😄

Reply:
For [these reasons](https://github.com/pipe-cd/pipe/issues/1595#issuecomment-784853228), we settled on disallowing changing git path anymore, that' why this PR opened.

And at the same time we have to guarantee it's immutable at DB level, that's what I mentioned on this PR body.

Reply:
> make it TODO:

That is so true. I'll add on another proper place.

Reply:
No, HERE is the proper place.

Reply:
thanks 😁